### ISSUE TITLE
Docker Hub Automated Build Status

### ISSUE THREAD

**Issue by bsdlp** [badges/shields#241]
There's already an [implementation](https://github.com/cpuguy83/docker-hub-buildtag) made by @cpuguy83, but it would be great if shields added support.

**Comment by noonien** [comment]
Any updates on this issue?

**Comment by cpuguy83** [comment]
No new api's on the hub yet.

**Comment by cybertk** [comment]
+1

**Comment by espadrine** [comment]
Do we have an example user/repo which is automated?

This kind of endpoint seems promising, but ubuntu isn't automated, I think: https://registry.hub.docker.com/v2/repositories/library/ubuntu/

**Comment by wblankenship** [comment]
They have been silently rolling out some metadata endpoints.

Example: https://registry.hub.docker.com/v2/repositories/wblankenship/dockeri.co/stars/count/

**Comment by hairyhenderson** [comment]
The docs are _partially_ here: http://docs.docker.com/reference/api/docker-io_api/, but they don't fully cover the `/v2/repositories/(repo)/` endpoint (mention of `GET` is missing...). I don't know if this means the API isn't public, or what... @cpuguy83?

Some other interesting endpoints:
- https://registry.hub.docker.com/v2/repositories/wblankenship/dockeri.co/ - gives _almost_ a full picture of the repo
- https://registry.hub.docker.com/v2/repositories/wblankenship/dockeri.co/tags/ - info about the tags, gives:
```json
{
  "count": 1,
  "tags": [
    "latest"
  ]
}
```

**Comment by wblankenship** [comment]
Just emailed support for documentation, will post back here with reply.

**Comment by korjavin** [comment]
Still nothing?

**Comment by cpuguy83** [comment]
It's not really documented yet, however some data is there now.
Hub 2.0 uses a pure client-side frontend, so all the api's the frontend needs to construct the page should be there.
Was able to find build history here:
```
curl -sSL -H 'Content-Type: application/json' https://hub.docker.com/v2/repositories/cpuguy83/ubuntu/buildhistory/
```
Looks like a build status of `10` is OK, `-1` is some failure, I don't know the in-betweens yet.
This should be better documented soon.
I wouldn't rely on this information for shields yet as it is undocumented, and now that Hub 2.0 is finally out, we'll start to see these API's exposed more.

**Comment by cpuguy83** [comment]
Also
 ```
curl -sSL -H 'Content-Type: application/json' https://hub.docker.com/v2/repositories/cpuguy83/ubuntu/autobuild/
```

**Comment by espadrine** [comment]
Thanks for this update!

**Comment by h0tbird** [comment]
+1

**Comment by p-a-c-o** [comment]
I found (using browser debug mode): 
- `tags` and then `tags/<name>`
- `build` or `autobuild`
- `dockerfile` 
- `comments`
- `buildhistory` and then `buildhistory/<build_code>`  <-- (includes logs and dockerfile)
- `links`
- `collaborators`
- `groups`
- `stars`
- `buildtrigger` (401) and `buildtrigger/history` (401)
- `webhooks` (401)

And extracted the status from the client.js:
`S={"-2":"exception","-1":"error",0:"pending",1:"claimed",2:"started",3:"cloned",4:"readme",5:"dockerfile",6:"built",7:"bundled",8:"uploaded",9:"pushed",10:"done"}
`

**Comment by hairyhenderson** [comment]
@p-a-c-o - nice!

Has anyone started on a PR for this?

**Comment by schmunk42** [comment]
:+1:

**Comment by espadrine** [comment]
Good work!

Unless someone makes a PR before then, I'll do this when I've processed the 60 remaining issues & PRs I haven't responded to yet.

**Comment by michaelmior** [comment]
@cpuguy83 I'm seeing status values of 3 for a build in progress and 0 for a build which is queued.

**Comment by eddiewebb** [comment]
Pretty small contribution, but I added the ability to report current status of last build.  Thanks @pl31 for the statuses!

10 = passing
< 0 = Failed
0-9 = Pending


<img width="849" alt="screen shot 2017-01-14 at 8 42 42 am" src="https://cloud.githubusercontent.com/assets/5262154/21955331/287954f6-da37-11e6-99dc-9bf58daee3bc.png">

**Comment by paulmelnikow** [comment]
Resolved via #856.
